contracts/AddressBook.sol

```diff
@@ -224,9 +224,9 @@ contract AddressBook is Ownable {
 
         if (proxyAddress == address(0)) {
             OwnedUpgradeabilityProxy proxy = new OwnedUpgradeabilityProxy();
-            proxy.upgradeToAndCall(_newAddress, params);
             setAddress(_id, address(proxy));
             emit ProxyCreated(_id, address(proxy));
+            proxy.upgradeToAndCall(_newAddress, params);
```

Comment:
why changing this here ?

Reply:
it's a better pattern to update storage data and then make external calls

Reply:
yeah to avoid reentrancy attack (y)